File: memento/src/main/java/com/alexstyl/specialdates/events/namedays/calendar/resource/Node.java
package com.alexstyl.specialdates.events.namedays.calendar.resource;

import com.alexstyl.specialdates.date.Date;
import com.alexstyl.specialdates.events.namedays.NameCelebrations;

public interface Node {

    void addDate(String name, Date date);

    NameCelebrations getDates(String name);
}

File: memento/src/main/java/com/alexstyl/specialdates/events/namedays/calendar/resource/Node.kt
package com.alexstyl.specialdates.events.namedays.calendar.resource

import com.alexstyl.specialdates.date.Date
import com.alexstyl.specialdates.events.namedays.NameCelebrations

interface Node {
    fun addDate(name: String, date: Date)
    fun getDates(name: String): NameCelebrations?
}
